doc/source/data/inspecting-data.rst

```diff
@@ -0,0 +1,130 @@
+.. _inspecting-data:
+
+===============
+Inspecting data
+===============
+
+.. _describing-datasets:
+
+Describing datasets
+===================
+
+:class:`Datasets <ray.data.Dataset>` are tabular. To view a Dataset's column names and
+types, call :meth:`Dataset.schema() <ray.data.Dataset.schema>`.
+
+.. testcode::
+
+    import ray
+
+    ds = ray.data.read_csv("s3://anonymous@air-example-data/iris.csv")
+
+    print(ds.schema())
+
+.. testoutput::
+
+    Column             Type
+    ------             ----
+    sepal length (cm)  double
+    sepal width (cm)   double
+    petal length (cm)  double
+    petal width (cm)   double
+    target             int64
+
+For more information such as the number of rows, print the Dataset.
+
+.. testcode::
+
+    import ray
+
+    ds = ray.data.read_csv("s3://anonymous@air-example-data/iris.csv")
+
+    print(ds)
+
+.. testoutput::
+
+    Dataset(
+       num_blocks=...,
+       num_rows=150,
+       schema={
+          sepal length (cm): double,
+          sepal width (cm): double,
+          petal length (cm): double,
+          petal width (cm): double,
+          target: int64
+       }
+    )
+
+.. _inspecting-rows:
+
+Inspecting rows
+===============
+
+To inspect rows, call :meth:`Dataset.take() <ray.data.Dataset.take>` or
+:meth:`Dataset.take_all() <ray.data.Dataset.take_all>`. Ray Data represents rows
+as dictionaries.
+
+.. testcode::
+
+    import ray
+
+    ds = ray.data.read_csv("s3://anonymous@air-example-data/iris.csv")
+
+    rows = ds.take(1)
+    print(rows)
+
+.. testoutput::
+
+    [{'sepal length (cm)': 5.1, 'sepal width (cm)': 3.5, 'petal length (cm)': 1.4, 'petal width (cm)': 0.2, 'target': 0}]
+
+
+For more information on working with rows, see
+:ref:`Transforming rows <transforming-rows>` and
+:ref:`Iterating over rows <iterating-over-rows>`.
+
+.. _inspecting-batches:
+
+Inspecting batches
+==================
+
+A batch contains data from multiple rows. To inspect batches, call
+`Dataset.take_batch() <ray.data.Dataset.take_batch>`. By default, Ray Data represents
+batches as ndarrays.
```

Comment:
Dictionary of Numpy ndarray. You can set the `"batch_format"` to change the format of the returned batch.